docs/specifications/tests/DNSSEC-TP/dnssec02.md

```diff
@@ -3,125 +3,202 @@
-   2. If no DNS response is returned, then output *[NO_RESPONSE]*
-      and go to next name server IP.
-   3. If the DNS response does not contain an DNSKEY RRset,
-      then output *[NO_RESPONSE_DNSKEY]* and go to next name server
-      IP.
-   4. Extract the DNSKEY RRset ("DNSKEY RRs").
-   5. If the DNS response does not contain any RRSIG records covering
-      the DNSKEY RRset, then output *[NO_RRSIG_DNSKEY]*.
-   5. Extract the RRSIG records covering the DNSKEY RRset, possibly
-      none ("DNSKEY RRSIG").
-   6. For each DS in *DS Records*, do:
-      1. Find the equivalent DNSKEY in *DNSKEY RRs* by key ID (key tag).
-      2. If matching DNSKEY is not found, output *[NO_MATCHING_DNSKEY]*.
-      3. If the DS values (algorithm and digest) do not match the
-         DNSKEY then output *[BROKEN_DS]*.
-      4. If bit 7 of the DNSKEY flags field has the value of 0 (nil),
-         then output *[DNSKEY_NOT_ZONE_SIGN]* and go to next DS.
-      5. If bit 15 of the DNSKEY flags field has the value of 0 (nil),
-         then output *[DNSKEY_KSK_NOT_SEP]*.
-      6. Find the equivalent RRSIG in *DNSKEY RRSIG* by key ID (key tag).
-      7. If matching RRSIG is not found, output *[NO_MATCHING_RRSIG]*.
-      8. If the RRSIG values (algorithm and signature) do not match
-         the DNSKEY then output *[BROKEN_RRSIG]*.
-
-8. If no message, besides possibly *[NO_RESPONSE]* or 
-   *[DNSKEY_KSK_NOT_SEP]*, has been outputted, then output 
-   *[DS_MATCHES]*.
+  (only if *Test Type* is [undelegated]).
+
+
+## Summary
+
+* If not both DS records and DNSKEY records are found, then further
+investigation will not be done and no messages will be outputted.
+* No messages will be outputted due to errors in the responses from the parent
+  name servers.
+
+Message Tag outputted              | Level   | Arguments          | Description of when message tag is outputted
+:----------------------------------|:--------|:-------------------|:--------------------------------------------
+DS02_NO_DNSKEY_FOR_DS              | ERROR   | ns_ip_list, keytag | The DNSKEY record that the DS record points to does not exist in the DNSKEY RRset.
+DS02_NO_MATCH_DS_DNSKEY            | ERROR   | ns_ip_list, keytag | The DS record does not match the DNSKEY record by algorithm or digest.
+DS02_DNSKEY_NOT_FOR_ZONE_SIGNING   | ERROR   | ns_ip_list, keytag | The DNSKEY record is not a key for DNSSEC signing.
+DS02_DNSKEY_NOT_SEP                | NOTICE  | ns_ip_list, keytag | The DNSKEY record that the DS refers to does not have the SEP bit set.
+DS02_NO_MATCHING_DNSKEY_RRSIG      | ERROR   | ns_ip_list, keytag | The DNSKEY RRset is not signed by the DNSKEY that the DS record refers to.
+DS02_ALGO_NOT_SUPPORTED_BY_ZM      | NOTICE  | ns_ip_list, algo_mnemo, algo_num, keytag | This installation of Zonemaster does not support the DNSKEY algorithm.
+DS02_RRSIG_NOT_VALID_BY_DNSKEY     | ERROR   | ns_ip_list, keytag | The RRSIG cannot be validated by the DNSKEY that it refers to.
```

Comment:
Could you sort the array by Message Tag?

Reply:
Fixed.